sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -198,7 +214,7 @@ object StateStore extends Logging {
   private def doMaintenance(): Unit = {
     logDebug("Doing maintenance")
     if (SparkEnv.get == null) {
```

Comment:
There is another potential issue here: if a SparkContext is created after checking `SparkEnv.get == null`, the following `stop` may cancel a new valid task. However, I think that won't happen in practice, so don't fix it.